pkgs/development/libraries/tdb/default.nix

```diff
@@ -50,4 +51,9 @@ stdenv.mkDerivation rec {
     license = licenses.lgpl3Plus;
     platforms = platforms.all;
   };
-}
+} // lib.optionalAttrs (stdenv.hostPlatform != stdenv.buildPlatform) {
+  # python-config from build Python gives incorrect values when cross-compiling.
+  # If python-config is not found, the build falls back to using the sysconfig
+  # module, which works correctly when cross-compiling.
+  PYTHON_CONFIG = "/invalid";
```

Comment:
Why not use optionalString?

Reply:
I'm not sure, I may have been trying to avoid rebuilding one of the native packages at one point. It appears that we can even avoid a special-casing cross-compilation at all, since the sysconfig fallback works for native builds as well. I would appreciate it if you could test `tdb` and `talloc` on darwin to make sure it works there too.